src/components/TimeBasedLineChart.js

```diff
@@ -42,713 +241,291 @@ class TimeBasedLineChart extends React.Component {
         adjustedWidth
       });
     }
-  }
-
-  shouldComponentUpdate (newProps, newState) {
-    return !isEqual(newProps.data, this.props.data) || !isEqual(newState.hoveredData, this.state.hoveredData);
-  }
-
-  _getAreaBelowZero (data) {
-    const area = d3.svg.area()
-      .x(d => {
-        const currentDate = moment.unix(d.timeStamp).startOf(this.props.rangeType).unix();
-
-        return this._getXScaleValue(currentDate);
-      })
-      .y0(this.state.adjustedHeight - 10)
-      .y1(this._getYScaleValue(0));
-
-    return area(data);
-  }
-
-  _getFlatLine (data) {
-    const flatLine = d3.svg.line()
-      .x(d => {
-        const currentDate = moment.unix(d.timeStamp).startOf(this.props.rangeType).unix();
-
-        return this._getXScaleValue(currentDate);
-      })
-      .y(this.state.adjustedHeight);
-
-    return flatLine(data);
-  }
-
-  _getLine (data) {
-    const line = d3.svg.line()
-      .x(d => {
-        const currentDate = moment.unix(d.timeStamp).startOf(this.props.rangeType).unix();
-
-        return this._getXScaleValue(currentDate);
-      })
-      .y(d => {
-        return this._getYScaleValue(d.value);
-      });
-
-    return line(data);
-  }
-
-  _getSliceMiddle () {
-    return this._getSliceWidth() / 2;
-  }
-
-  _getSliceWidth () {
-    return Math.floor(this.state.adjustedWidth / this.props.data.length);
-  }
-
-  _getSplitData () {
-    const future = [];
-    const past = [];
-
-    this.props.data.forEach(item => {
-      const currentDate = moment.unix(item.timeStamp).startOf(this.props.rangeType);
-
-      if (currentDate.isBefore(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
-        past.push(item);
-      }
+  },
 
-      if (currentDate.isAfter(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
-        future.push(item);
-      }
+  componentDidUpdate () {
+    this._styleChart();
+  },
 
-      if (currentDate.isSame(moment.unix(this.props.breakPointDate), this.props.rangeType)) {
-        past.push(item);
-        future.push(item);
-      }
+  // Handle functions
+  _handleChartMouseLeave () {
+    this.setState({
+      hoveredDataPoint: null
     });
+  },
 
-    return [past, future];
-  }
-
-  _getYAxisTicks (data, numTicks = 4) {
-    const maxValue = d3.max(this.props.data, d => {
-      const value = d.above ? d.value + d.above : d.value + 1000;
-      const multiplier = value < 0 ? -1000 : 1000;
-
-      return Math.ceil(value / multiplier) * multiplier;
+  _handleChartMouseOver (hoveredDataPoint) {
+    this.setState({
+      hoveredDataPoint
     });
+  },
 
-    let minValue = d3.min(this.props.data, d => {
-      const value = d.below ? d.value + d.below : d.value - 1000;
-      const multiplier = value < 0 ? -1000 : 1000;
+  // Helper Functions
+  _getFormatedValue (value, type, format) {
```

Comment:
No biggie but you may want to fix the spelling here. It should be `_getFormattedValue`